scripts/generate_drivers.py

```diff
@@ -186,10 +197,43 @@ def generate_metal_dev_hdr(vendor, device, index, reglist):
     #include <metal/compiler.h>
     #include <metal/io.h>
 
+    // Private utility functions
+
+    // Write data into register field by only changing bits within that field.
+    static inline void write_field(
+        volatile uint32_t *register_base,
+        uint32_t field_offset,
+        uint32_t field_width,
+        uint32_t field_data
+    ) {
+        const uint32_t shifted_field_data = field_data << field_offset;
+        const uint32_t mask = (field_width == 32) ? 0xffffffff : ((1 << field_width) - 1) << field_offset;
+        const uint32_t original_data = *register_base;
+
+        const uint32_t cleared_data = original_data & (~mask);
+        const uint32_t new_data = cleared_data | shifted_field_data;
```

Comment:
Nice to see the younger generation understanding C bit twiddling.

Reply:
C is still taught in schools!

I will admit, however, that functional programming has poisoned my brain and now I try declaring as many variables as possible as `const` in C.